lib/openssl/x509.rb

```diff
@@ -60,6 +60,109 @@ def to_h # {"oid"=>sn|ln, "value"=>value, "critical"=>true|false}
       def to_a
         [ self.oid, self.value, self.critical? ]
       end
+
+      module Helpers
+        def find_extension(oid)
+          extensions.find { |e| e.oid == oid }
+        end
+
+        def x509_name_from_general_name(gn_asn1)
+          unless gn_asn1.tag_class == :CONTEXT_SPECIFIC
+            raise ASN1::ASN1Error "invalid GeneralName"
+          end
+
+          if gn_asn1.tag == 4
```

Comment:
What is 4?

Reply:
In this case, it designates that this is a `directoryName` (see [RFC 5280 Appendix A.2](https://tools.ietf.org/html/rfc5280#appendix-A.2)), which is a `Name` (see [RFC 5280 Section 4.1.2.4](https://tools.ietf.org/html/rfc5280#section-4.1.2.4))

```
GeneralName ::= CHOICE {
     otherName                 [0]  AnotherName,
     rfc822Name                [1]  IA5String,
     dNSName                   [2]  IA5String,
     x400Address               [3]  ORAddress,
     directoryName             [4]  Name,
     ediPartyName              [5]  EDIPartyName,
     uniformResourceIdentifier [6]  IA5String,
     iPAddress                 [7]  OCTET STRING,
     registeredID              [8]  OBJECT IDENTIFIER }

Name ::= CHOICE { -- only one possibility for now --
     rdnSequence  RDNSequence }

RDNSequence ::= SEQUENCE OF RelativeDistinguishedName

RelativeDistinguishedName ::= SET SIZE (1..MAX) OF AttributeTypeAndValue

AttributeTypeAndValue ::= SEQUENCE {
     type     AttributeType,
     value    AttributeValue }

AttributeType ::= OBJECT IDENTIFIER

AttributeValue ::= ANY -- DEFINED BY AttributeType
```

Reply:
Do you think it makes sense to add some constants?